source/extensions/quic_listeners/quiche/codec_impl.h

```diff
@@ -55,5 +57,45 @@ class QuicHttpServerConnectionImpl : public QuicHttpConnectionImplBase,
   EnvoyQuicServerSession& quic_server_session_;
 };
 
+class QuicHttpClientConnectionImpl : public QuicHttpConnectionImplBase,
+                                     public Http::ClientConnection {
+public:
+  QuicHttpClientConnectionImpl(EnvoyQuicClientSession& session,
+                               Http::ConnectionCallbacks& callbacks);
+
+  // Http::ClientConnection
+  Http::StreamEncoder& newStream(Http::StreamDecoder& response_decoder) override;
+
+  // Http::Connection
+  void goAway() override { NOT_REACHED_GCOVR_EXCL_LINE; }
```

Comment:
Q: I assume HTTP/3 does support this same dance? Do we have to implement this eventually? If so TODO? And I guess then the HCM code uses an HTTP/2 only check for this type of goaway/drain sequence?

Reply:
This client codec. Client shouldn't send GOAWAY according to wire spec.

Reply:
Oops yup.